File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        
        StringBuilder string = new StringBuilder();

        for(char c : s.toCharArray()){
            if(Character.isLetterOrDigit(c)){
                string.append(Character.toLowerCase(c));
            }
        }

        int i = 0;
        int j = string.length()-1;

        while(i < j){
            if(string.charAt(j) != string.charAt(i)){
                return false;
            }
            i++;
            j--;
        }
        return true;
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution(object):
    def isPalindrome(self, s):
        new_str = ""

        for ch in s:
            if ch.isalnum():
                new_str += ch.lower()

        i = 0
        j = len(new_str) - 1

        while i < j:
            if new_str[i] != new_str[j]:
                return False 
            i += 1
            j -= 1

        return True
        